pkg/dependencies/factory.go

```diff
@@ -1103,6 +1104,36 @@ func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster, timeoutOpt
 	return f
 }
 
+// WithEKSAInstaller builds a cluster manager based on the cluster config and timeout options.
+func (f *Factory) WithEKSAInstaller() *Factory {
+	f.WithClusterctl().WithKubectl().WithFileReader()
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		if f.dependencies.EksaInstaller != nil {
+			return nil
+		}
+
+		var r *retrier.Retrier
+		if f.config.noTimeouts {
+			r = retrier.NewWithNoTimeout()
+		} else {
+			r = clustermanager.DefaultRetrier()
+		}
+
+		client := clustermanager.NewRetrierClient(
+			f.dependencies.Kubectl,
+			r,
+		)
```

Comment:
it seems like we could move this logic to a `WithKubernetesRetrierClient` and reuse it from both the cluster manager constructor and this one